server/users/users.service.ts

```diff
@@ -13,25 +14,30 @@ export class UsersService {
     ) {}
 
     async findAll(getUsers): Promise<User[]> {
-        const {searchName} = getUsers;
+        const { searchName } = getUsers;
         return this.UserModel.aggregate([
-            { $match: {name: { '$regex': searchName, '$options': 'i' }}},
-            { $project: { "_id": 1, "name": 1}}
-        ])
+            { $match: { name: { $regex: searchName, $options: "i" } } },
+            { $project: { _id: 1, name: 1 } },
+        ]);
     }
 
     async register(user) {
-        const newUser = new this.UserModel(user)
-        const { data: oryUser } = await this.oryService.createIdentity(newUser, user.password);
-        newUser.oryId = oryUser.id;
-        try {
-            // @ts-ignore
-            return this.UserModel.register(
+        const newUser = new this.UserModel(user);
+
+        if (!newUser.oryId) {
+            const { data: oryUser } = await this.oryService.createIdentity(
                 newUser,
                 user.password
             );
+            newUser.oryId = oryUser.id;
+        }
+        try {
+            // @ts-ignore
+            return this.UserModel.register(newUser, user.password);
         } catch (e) {
-            await this.oryService.deleteIdentity(oryUser.id)
+            if (!newUser.oryId) {
+                await this.oryService.deleteIdentity(newUser.oryId);
```

Comment:
We can stop doing this considering that we will be creating some users outside the platform through Ory Cloud.